clientd3d/d3drender.c

```diff
@@ -6909,14 +6874,31 @@ void D3DRenderObjectsDraw(d3d_render_pool_new *pPool, room_type *room,
 		if (pRNode->obj.id == player.id)
 			continue;
 
-		if (flags & OF_INVISIBLE)
+		if (processedIds.find(pRNode->obj.id) == processedIds.end())

```

Comment:
Another option would be to insert all the objects in a set before the loop to remove duplicates, then iterate over the set. 
 Then you wouldn't need to check each time through the loop.

But even if not, rewrite to the clearer:
```
if (processedIds.find(pRNode->obj.id) != processedIds.end())
  continue;
processedIds.insert(pRNode->obj.id);
```